psdk_wrapper/src/modules/flight_control.cpp

```diff
@@ -903,10 +903,236 @@ void
 FlightControlModule::flight_control_generic_cb(
     const sensor_msgs::msg::Joy::SharedPtr msg)
 {
-  /** @todo implemnent generic control functionality */
+  /** @author Umesh Mane
+   * Contact: umeshmane280@gmail.com
+   */
+
   (void)msg;
-  RCLCPP_WARN(get_logger(),
-              "Generic control setpoint is not currently implemented!");
+  float x_setpoint = msg->axes[0];
+  float y_setpoint = msg->axes[1];
+  float z_setpoint = msg->axes[2];
+  float yaw_setpoint = (uint8_t)(msg->axes[3]);
+  uint8_t flag = msg->axes[4];
+
+  uint8_t HORI = (flag & 0xC0);
+  uint8_t VERT = (flag & 0x30);
+  uint8_t YAW = (flag & 0x08);
+  uint8_t FRAME = (flag & 0x06);
+  uint8_t HOLD = (flag & 0x01);
+
+  float x_cmd, y_cmd, z_cmd, yaw_cmd;
+  if (FRAME == Control::FRAME_GROUND)
+  {
+    FG = true;
```

Comment:
do we really need these additional variables FG, FB? Couldn't you already set the joystick mode directly here? Or is there something I am missing